lib/ansible/modules/packaging/os/yum.py

```diff
@@ -33,78 +33,87 @@
         several packages this can accept a comma separated list of packages
         or (as of 2.0) a list of packages.
     required: true
-    aliases: [ pkg ]
+    aliases: ['pkg']
   exclude:
     description:
-      - "Package name(s) to exclude when state=present, or latest"
-    version_added: "2.0"
+      - 'Package name(s) to exclude when state=present, or latest'
+    version_added: '2.0'
   list:
     description:
-      - "Package name to run the equivalent of yum list <package> against. In addition to listing packages,
-        use can also list the following: C(installed), C(updates), C(available) and C(repos)."
+      - 'Package name to run the equivalent of yum list <package> against. In addition to listing packages,
+        use can also list the following: C(installed), C(updates), C(available) and C(repos).'
   state:
     description:
       - Whether to install (C(present) or C(installed), C(latest)), or remove (C(absent) or C(removed)) a package.
-    choices: [ absent, installed, latest, present, removed ]
+    choices: ['absent', 'installed', 'latest', 'present', 'removed']
     default: present
+  autoremove:
```

Comment:
This should either have a note about the version of yum required to make this work (the version of yum in RHEL/CentOS6 does not have this functionality) or implement it in a way that makes use of `clean_requirements_on_remove` instead of the `autoremove` yum sub-command.

If you do go the route of note about version requirements, I think it would be a good idea to have a check in the code that will do something like `module.fail_json(msg="YUM Version {} Required for autoremove".format(YUM_MIN_VERSION))` if an older version is found.